app/airflow/dags/libs/SR_processing/helpers.py

```diff
@@ -0,0 +1,183 @@
+    app/shared/services/utils.py -> function: process_four_item_dict
+    """
+    csv_file_names = set(row["csv_file_name"] for row in four_item_data)
+    # Initialise the dictionary with the keys, and each value set to a blank dict()
+    new_data_dictionary: Dict[str, Dict[str, Dict[str, str]]] = {}
+    for csv_file_name in csv_file_names:
+        new_data_dictionary[csv_file_name] = {}
+
+    for row in four_item_data:
+        if row["field_name"] not in new_data_dictionary[row["csv_file_name"]]:
+            new_data_dictionary[row["csv_file_name"]][row["field_name"]] = {}
+        new_data_dictionary[row["csv_file_name"]][row["field_name"]][row["code"]] = row[
+            "value"
+        ]
+
+    return new_data_dictionary
+
+
+def transform_scan_report_sheet_table(sheet: Worksheet) -> defaultdict[Any, List]:
+    """
+    Transforms a worksheet data into a JSON like format.
+    Note: This function was copied from the workers project. More details can be found here:
+    app/workers/UploadQueue/__init__.py -> function: _transform_scan_report_sheet_table
+
+    Args:
+        sheet (Worksheet): Sheet of data to transform
+
+    Returns:
+        defaultdict[Any, List]: The transformed data.
+    """
+    logging.debug("Start process_scan_report_sheet_table")
+
+    sheet.calculate_dimension()
+    # Get header entries (skipping every second column which is just 'Frequency')
+    # So sheet_headers = ['a', 'b']
+    first_row = sheet[1]
+    sheet_headers = [cell.value for cell in first_row[::2]]
+
+    d = defaultdict(list)
+    for row in sheet.iter_rows(
+        min_col=1,
+        max_col=len(sheet_headers) * 2,
+        min_row=2,
+        max_row=sheet.max_row,
+        values_only=True,
+    ):
+        # Set boolean to track whether we hit a blank row for early exit below.
+        this_row_empty = True
+        # Iterate across the pairs of cells in the row. If the pair is non-empty,
+        # then add it to the relevant dict entry.
+        for header, cell, freq in zip(sheet_headers, row[::2], row[1::2]):
+            if (cell != "" and cell is not None) or (freq != "" and freq is not None):
+                d[header].append((str(cell), freq))
+                this_row_empty = False
+        if this_row_empty:
+            break
+    # Clean BOM characters from keys before returning
+    cleaned_dict = defaultdict(list)
+    for key, value in d.items():
+        clean_key = key.replace("\ufeff", "") if isinstance(key, str) else key
+        cleaned_dict[clean_key] = value
+
+    logging.debug("Finish process_scan_report_sheet_table")
+    return cleaned_dict
+
+
+def default_zero(value) -> float:
+    """
+    Helper function that returns the input, replacing anything Falsey
+    (such as Nones or empty strings) with 0.0.
+    """
+    return round(value or 0.0, 2)
+
+
+# Storage type
+storage_type = os.getenv("STORAGE_TYPE", StorageType.MINIO)
```

Comment:
Haven't seen any others yet - but at some point it's worth following a pattern and importing all config into a `settings.py` to centralise it